frontend/catalyst/pennylane_extensions.py

```diff
@@ -110,20 +111,38 @@ class QFunc:
     """
 
     def __init__(self, fn, device):
-        self.func = fn
-        self.device = device
-        update_wrapper(self, fn)
+        self.func = fn  # pragma: nocover
+        self.device = device  # pragma: nocover
+        update_wrapper(self, fn)  # pragma: nocover
+
+    @staticmethod
+    def _add_toml_file(device):
+        """Temporary function. This function adds the config field to devices.
+        TODO: Remove this function when `qml.Device`s are guaranteed to have their own
+        config file field."""
+        device_lpath = pathlib.Path(get_lib_path("runtime", "RUNTIME_LIB_DIR"))
+        name = device.name
+        if isinstance(device, qml.Device):
+            name = device.short_name
+
+        # The toml files name convention we follow is to replace
+        # the dots with underscores in the device short name.
+        toml_file_name = name.replace(".", "_") + ".toml"
+        # And they are currently saved in the following directory.
+        toml_file = device_lpath.parent / "lib" / "backend" / toml_file_name
+        device.config = toml_file
 
     def __call__(self, *args, **kwargs):
         qnode = None
-        if isinstance(self, QNode):
+        if isinstance(self, qml.QNode):  # pragma: nocover
             qnode = self
+            QFunc._add_toml_file(self.device)
             device = QJITDevice(
                 self.device.shots, self.device.wires, *extract_backend_info(self.device)
             )
-        else:
+        else:  # pragma: nocover
             # Allow QFunc to still be used by itself for internal testing.
-            device = self.device
+            device = self.device  # pragma: nocover
```

Comment:
Quick side question: I thought we did execute this function, is the coverage package just not detecting it because we patch it in, or did we change this behaviour?

This one shouldn't be required since the block is marked:
```suggestion
            device = self.device
```

Reply:
This is tested but it is tested in lit tests. Lit tests are not part of coverage.

Reply:
[nocover](https://github.com/PennyLaneAI/catalyst/pull/369/commits/cf62d0901e31ce750338bf235499a3b7dff65696) Will double check once CI/CD runs.

Reply:
Wrong link I think, that's for the init function